packages/nextjs/src/config/wrappers/wrapperUtils.ts

```diff
@@ -1,12 +1,119 @@
-import { captureException } from '@sentry/core';
+import { captureException, getCurrentHub, startTransaction } from '@sentry/core';
 import { getActiveTransaction } from '@sentry/tracing';
+import { Transaction } from '@sentry/types';
+import { fill } from '@sentry/utils';
+import * as domain from 'domain';
+import { IncomingMessage, ServerResponse } from 'http';
+
+declare module 'http' {
+  interface IncomingMessage {
+    _sentryTransaction?: Transaction;
+  }
+}
+
+function getTransactionFromRequest(req: IncomingMessage): Transaction | undefined {
+  return req._sentryTransaction;
+}
+
+function setTransactionOnRequest(transaction: Transaction, req: IncomingMessage): void {
+  req._sentryTransaction = transaction;
+}
+
+function autoEndTransactionOnResponseEnd(transaction: Transaction, res: ServerResponse): void {
+  fill(res, 'end', (originalEnd: ServerResponse['end']) => {
+    return function (this: unknown, ...endArguments: Parameters<ServerResponse['end']>) {
+      transaction.finish();
+      return originalEnd.call(this, ...endArguments);
+    };
+  });
+}
+
+/**
+ * Wraps a function that potentially throws. If it does, the error is passed to `captureException` and retrhrown.
```

Comment:
```suggestion
 * Wraps a function that potentially throws. If it does, the error is passed to `captureException` and rethrown.
```

Am I missing the rethrowing? I don't see it here.

Reply:
Yeah whoops. Thanks! 5970fede5e61973df0fb287387f93fdea639fe57